nasa/ow_simulator
# OCEANWATER-581 Power Fault Impact Ros Topics

## Linked Issues:
| Jira Ticket 🎟️   | [Oceanwater-581](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-581)  [Oceanwater-622](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-622)|
| ----------- | ----------- |
| EPIC ⚡| [Oceanwater-551](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-551) |
| Github :octocat:  | #111 (cleanup) |

Please Note: 
This PR will act as the base PR for a series of power faults prs. 

## Summary of Changes
*  faults node published a trigger topic when users select /rqt
** low volt = 1, cap loss = 2, thermal = 4 (in bits)
** power node listens to this topic /faults/power_faults
* power node sets node booleans to determine which power fault has been triggered
** new power node function sets the new value of the mechanical power depending on which fault is chosen
* faults node listens for selection of any fault and checks SOC values and TEMP values to publish power faults message to topic /power_faults_status

## Test

[Oceanwater-581](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-581)  
 TEST RQT FAULT TRIGGER:
* see Ussama's power PR #113 

[Oceanwater-622](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-622)
TEST FAULTS OUTPUTTING TRIGGERS FOR  LOW VOLTAGE / CAP LOSS / THERMAL 
* start any launch file
* `rostopic echo /faults/jpl/power_faults_status`
* see value as hardware fault (1) for any power fault

[future PR, do not test yet]
TEST REAL POWER OUTPUT 
* roslaunch any launch file
* select any:
** `rostopic echo /power_system_node/state_of_charge`
*** for low volt, value < 10
*** for cap loss, value difference should be > 5%
** `rostopic echo /power_system_node/battery_temperature`
*** temperature must be > 50C
